new_platforms/docs/visualstudio_dev.md

```diff
@@ -0,0 +1,109 @@
+
+**To create your own enclave:**
+1. In Visual Studio, add a new Visual C++ "Open Enclave TEE Project".
+2. Edit the _YourProjectName_.edl file. Define any trusted APIs (called "ECALLs")
+you want to call from your application in the trusted{} section,
+and in the untrusted{} section, define any application APIs (called "OCALLs")
+that you want to call from your enclave. Definitions must be done using the
+[EDL file syntax](https://software.intel.com/en-us/sgx-sdk-dev-reference-enclave-definition-language-file-syntax).
+3. Edit the ecalls.c file, and fill in implementations of the ECALL(s) you added.
+
+**To call your enclave from an application project:**
+1. In Visual Studio, create a new Visual C++ project, or open an existing
+one, that will build a normal application that will call your enclave APIs.
+2. Right click on your application project, select
+"Open Enclave Configuration"->"Import Enclave", and select the
+_YourProjectName_.edl file from your enclave project.
+3. Add code in your app to call oe\_create\__YourEDLFileName_\_enclave(),
+any ECALLs you added, and
+oe\_terminate\_enclave().  You will need to #include <openenclave/host.h> 
+and <_YourEDLFileName_\_u.h> for your ECALLs.
+See the sample apps for an example.
+
+OP-TEE only allows one thread per TA to be in an ECALL (i.e., a call into
+a TA from a host app).  Even if it has an OCALL (i.e., an out-call
+back into the host app) in progress, the ECALL must complete before
+another ECALL can enter the TA.  SGX, on the other hand, would allow a
+second ECALL to enter.  So if you want them to function identically, host apps
+can pass the OE\_ENCLAVE\_FLAG\_SERIALIZE\_ECALLS
+flag when creating an enclave to automatically get the OP-TEE like behavior
+for both SGX and TrustZone.
+
+
+**Then to build for OP-TEE:**
+
+1. Right click the solution, select "Configuration Manager", create an ARM
+solution platform (if one doesn't already exist), and then create ARM
+configurations of your application project (and oehost if you include that
+directly).  Copy the configuration from the existing Win32 one.  Don't do
+this for your enclave project, as that needs to be built from a bash shell
+rather than in Visual Studio.
+2. Manually edit your application .vcxproj file to add the ability to
+compile for ARM, since Visual Studio cannot do it from the UI.  To do so, add the
+line "<WindowsSDKDesktopARMSupport\>true</WindowsSDKDesktopARMSupport\>"
+to each ARM configuration property group.  (See the sample apps'
+vcxproj file for examples.)
+3. For any new source files you add to your enclave project in
+Visual Studio, also add them to the sub.mk file in your optee subdirectory
+4. In your application project properties, under "Linker"->"Input", add
+rpcrt4.lib to the Additional Dependencies (All Configurations, ARM platform)
+which is required for string-to-UUID conversion, and remove any sgx libs.
+5. In your application project properties, update the Additional Include
```

Comment:
Adding the additional include directories, libraries, and additional library directories should be done by the `targets` file in the NuGet package. For a new host project, the additional include directories, libraries, and additional library directories should be empty and the `targets` file should pick it up. For existing projects, it should be sufficient to clear out these lists for for both OP-TEE and SGX (except for any additions made by the developer to other dependencies).

I should be able to add `rpcrt4.lib` as an additional dependency when targetting OP-TEE.

Reply:
Great.  OP-TEE support in the templates is not yet complete, as noted in the PR.